### PR TITLE
Improve epoch_result_store code quality

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Fixes many code smells I found while debugging #4863
No logic changes.
I suggest reviewing the first commit separately. The other commit is just `black` changes.

## Before submitting
- [n/a] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [n/a] Did you make sure to update the documentation with your changes?
- [n/a] Did you write any new necessary tests? 
- [x] Did you verify new and existing tests pass locally with your changes?
- [n/a] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified; _Bugfixes should be including in bug-fix release milestones (m.f.X) and features should be included in (m.X.b) releases._

### CODE REVIEW

File: pytorch_lightning/trainer/connectors/logger_connector/epoch_result_store.py [hunk 1]
```diff
@@ -462,11 +388,9 @@ def update_logger_connector(self, fx_name: str = None) -> None:
...
 
     def run_batch_from_func_name(self, func_name) -> Dict:
-        results = []
-        for fx_name, hook_result in self._internals.items():
-            func = getattr(hook_result, func_name)
-            results.append(func(include_forked_originals=False))
-        return dict(ChainMap(*sum(results, [])))
+        results = [getattr(hook_result, func_name) for hook_result in self._internals.values()]
+        results = [func(include_forked_originals=False) for func in results]
+        return {k: v for d in sum(results, []) for k, v in d.items()}  # List[List[dict]] -> dict
```
Comment (justusschock) [inline pytorch_lightning/trainer/connectors/logger_connector/epoch_result_store.py:393]: I'd assume a chain map to be more efficient than dict comprehensions, no?

**Review by justusschock — COMMENTED** [review]
I really like it. So far I only have one minor question regarding performance.

Probably @tchaton is the best one to review this!

**Review by tchaton — APPROVED** [review]
Awesome cleanup ! Thx.

File: pytorch_lightning/trainer/connectors/logger_connector/epoch_result_store.py (hunk 1 above)
Comment (tchaton) [reply pytorch_lightning/trainer/connectors/logger_connector/epoch_result_store.py:393]: I have done some tests and ChainMap weren't faster. I am good with this 👍

**Review by justusschock — APPROVED** [review]

**Review by SeanNaren — APPROVED** [review]

**Pull request merged**